Mathlib/RingTheory/OreLocalization/Basic.lean

```diff
@@ -266,81 +281,100 @@ protected def smul : R[S⁻¹] → X[S⁻¹] → X[S⁻¹] :=
     rw [mul_assoc _ r₃', ← h₃, ← mul_assoc, ← mul_assoc]
 #align ore_localization.mul OreLocalization.smul
 
+@[to_additive]
 instance instSMulOreLocalization : SMul R[S⁻¹] X[S⁻¹] :=
   ⟨OreLocalization.smul⟩
 
+@[to_additive]
 instance instMulOreLocalization : Mul R[S⁻¹] :=
   ⟨OreLocalization.smul⟩
 
+@[to_additive]
 theorem oreDiv_smul_oreDiv {r₁ : R} {r₂ : X} {s₁ s₂ : S} :
     (r₁ /ₒ s₁) • (r₂ /ₒ s₂) = oreNum r₁ s₂ • r₂ /ₒ (oreDenom r₁ s₂ * s₁) :=
   rfl
 
+@[to_additive]
 theorem oreDiv_mul_oreDiv {r₁ : R} {r₂ : R} {s₁ s₂ : S} :
     (r₁ /ₒ s₁) * (r₂ /ₒ s₂) = oreNum r₁ s₂ * r₂ /ₒ (oreDenom r₁ s₂ * s₁) :=
   rfl
 #align ore_localization.ore_div_mul_ore_div OreLocalization.oreDiv_mul_oreDiv
 
 /-- A characterization lemma for the scalar multiplication on the Ore localization,
 allowing for a choice of Ore numerator and Ore denominator. -/
+@[to_additive "A characterization lemma for the vector addition on the Ore localization,
+allowing for a choice of Ore minuend and Ore subtrahend."]
 theorem oreDiv_smul_char (r₁ : R) (r₂ : X) (s₁ s₂ : S) (r' : R) (s' : S) (huv : s' * r₁ = r' * s₂) :
     (r₁ /ₒ s₁) • (r₂ /ₒ s₂) = r' • r₂ /ₒ (s' * s₁) :=
   smul'_char r₁ r₂ s₁ s₂ s' r' huv
 
 /-- A characterization lemma for the multiplication on the Ore localization, allowing for a choice
 of Ore numerator and Ore denominator. -/
+@[to_additive "A characterization lemma for the addition on the Ore localization,
+allowing for a choice of Ore minuend and Ore subtrahend."]
 theorem oreDiv_mul_char (r₁ r₂ : R) (s₁ s₂ : S) (r' : R) (s' : S) (huv : s' * r₁ = r' * s₂) :
     r₁ /ₒ s₁ * (r₂ /ₒ s₂) = r' * r₂ /ₒ (s' * s₁) :=
   smul'_char r₁ r₂ s₁ s₂ s' r' huv
 #align ore_localization.ore_div_mul_char OreLocalization.oreDiv_mul_char
 
 /-- Another characterization lemma for the scalar multiplication on the Ore localizaion delivering
 Ore witnesses and conditions bundled in a sigma type. -/
+@[to_additive "Another characterization lemma for the vector addition on the
+  Ore localizaion delivering Ore witnesses and conditions bundled in a sigma type."]
 def oreDivSMulChar' (r₁ : R) (r₂ : X) (s₁ s₂ : S) :
     Σ'r' : R, Σ's' : S, s' * r₁ = r' * s₂ ∧ (r₁ /ₒ s₁) • (r₂ /ₒ s₂) = r' • r₂ /ₒ (s' * s₁) :=
   ⟨oreNum r₁ s₂, oreDenom r₁ s₂, ore_eq r₁ s₂, oreDiv_smul_oreDiv⟩
 
 /-- Another characterization lemma for the multiplication on the Ore localizaion delivering
 Ore witnesses and conditions bundled in a sigma type. -/
+@[to_additive "Another characterization lemma for the addition on the Ore localizaion delivering
+  Ore witnesses and conditions bundled in a sigma type."]
 def oreDivMulChar' (r₁ r₂ : R) (s₁ s₂ : S) :
     Σ'r' : R, Σ's' : S, s' * r₁ = r' * s₂ ∧ r₁ /ₒ s₁ * (r₂ /ₒ s₂) = r' * r₂ /ₒ (s' * s₁) :=
   ⟨oreNum r₁ s₂, oreDenom r₁ s₂, ore_eq r₁ s₂, oreDiv_mul_oreDiv⟩
 #align ore_localization.ore_div_mul_char' OreLocalization.oreDivMulChar'
 
+@[to_additive]
 instance instOneOreLocalization : One R[S⁻¹] :=
```

Comment:
This instance' name becomes `instZeroOreLocalization` rather than `instZeroAddOreLocalization`: can the second be forced?

Reply:
Goot catch. Fixed. I think this example is particularly confusing for `to_additive` because `OreLocalization` is mixed into a longer camel-case name. This shouldn't happen that often.

Reply:
Well, unfortunately it happens all over the file (see `oreDiv` or `oreDivSmulChar`, etc...). I do not think it is reasonable to fix all them by hand, or at least it will certainly be painful. That's why I was asking for a reference in the library to see how this kind of things are handled. I am happy to ask on Zulip if you prefer. And I am sorry for making this review so long, it should have been an "easy chore splitting..." ;-)

Reply:
I think the generated `AddOreLocalization.oreSubAddChar'` is the correct name?

Reply:
Why not `AddOreLocalization.addoreSubAddChar'`?

Reply:
If the instance got changed from `OreLocalization` to `AddOreLocalization` this should be consistent. Or perhaps not. Or we could remove all these `ore` and possibly make something `private`?